diff --git a/arch/arm/src/stm32h7/stm32_start.c b/arch/arm/src/stm32h7/stm32_start.c
index 60eef6a85d..4f11cf28c2 100644
--- a/arch/arm/src/stm32h7/stm32_start.c
+++ b/arch/arm/src/stm32h7/stm32_start.c
@@ -272,6 +272,7 @@ void __start(void)
 
 #ifdef CONFIG_ARCH_PERF_EVENTS
   up_perf_init((void *)STM32_CPUCLK_FREQUENCY);
+  showprogress('P');
 #endif
 
   /* Perform early serial initialization */
