diff --git a/arch/arm/src/armv7-m/arm_perf.c b/arch/arm/src/armv7-m/arm_perf.c
index a8b58c5e..3ee8fa5e 100644
--- a/arch/arm/src/armv7-m/arm_perf.c
+++ b/arch/arm/src/armv7-m/arm_perf.c
@@ -46,18 +46,39 @@ static unsigned long g_cpu_freq = ULONG_MAX;
 
 void up_perf_init(void *arg)
 {
+#ifdef CONFIG_DEBUG_FEATURES
+  arm_lowputc('F');
+#endif
+
   g_cpu_freq = (unsigned long)(uintptr_t)arg;
 
+#ifdef CONFIG_DEBUG_FEATURES
+  arm_lowputc('G');
+#endif
+
   /* Enable ITM and DWT resources, if not left enabled by debugger. */
 
   modifyreg32(NVIC_DEMCR, 0, NVIC_DEMCR_TRCENA);
 
+#ifdef CONFIG_DEBUG_FEATURES
+  arm_lowputc('R');
+#endif
+
   /* Make sure the high speed cycle counter is running.  It will be started
    * automatically only if a debugger is connected.
    */
 
   putreg32(0xc5acce55, ITM_LAR);
+
+#ifdef CONFIG_DEBUG_FEATURES
+  arm_lowputc('I');
+#endif
+
   modifyreg32(DWT_CTRL, 0, DWT_CTRL_CYCCNTENA_MASK);
+
+#ifdef CONFIG_DEBUG_FEATURES
+  arm_lowputc('W');
+#endif
 }
 
 unsigned long up_perf_getfreq(void)
